components/feature/webnotifications/src/main/java/mozilla/components/feature/webnotifications/NativeNotificationBridge.kt

```diff
@@ -18,16 +21,23 @@ import mozilla.components.browser.icons.IconRequest.Size
 import mozilla.components.concept.engine.webnotifications.WebNotification
 
 internal class NativeNotificationBridge(
-    private val icons: BrowserIcons
+    private val icons: BrowserIcons,
+    @DrawableRes private val smallIcon: Int
 ) {
+    companion object {
+        private const val EXTRA_ON_CLICK = "mozac.feature.webnotifications.generic.onclick"
+    }
 
     /**
      * Create a system [Notification] from this [WebNotification].
      */
+    @Suppress("LongParameterList")
     suspend fun convertToAndroidNotification(
         notification: WebNotification,
         context: Context,
-        channelId: String
+        channelId: String,
+        activityClass: Class<*>?,
```

Comment:
This wildcard seems too permissive. Maybe `Class<out Activity>?`?

Reply:
Agreed.

Reply:
Agreed.